x-pack/platform/plugins/private/observability_ai_assistant_management/public/routes/components/settings_tab/change_kb_model.test.tsx

```diff
@@ -0,0 +1,235 @@
+  ...jest.requireActual('@kbn/observability-ai-assistant-plugin/public'),
+  useKibana: () => ({
+    services: {
+      overlays: {
+        openConfirm: jest.fn(() => Promise.resolve(true)),
+      },
+    },
+  }),
+}));
+
+jest.mock('@kbn/ai-assistant/src/utils/get_model_options_for_inference_endpoints');
+
+const mockGetModelOptions = jest.mocked(modelOptionsModule.getModelOptionsForInferenceEndpoints);
+
+const createMockStatus = (
+  overrides?: Partial<APIReturnType<'GET /internal/observability_ai_assistant/kb/status'>>
+): UseKnowledgeBaseResult['status'] => ({
+  value: {
+    enabled: true,
+    kbState: KnowledgeBaseState.READY,
+    isReIndexing: false,
+    currentInferenceId: ELSER_IN_EIS_INFERENCE_ID,
+    concreteWriteIndex: 'index_1',
+    endpoint: {
+      inference_id: ELSER_IN_EIS_INFERENCE_ID,
+      task_type: 'text_embedding',
+      service: 'my-service',
+      service_settings: {},
+    },
+    ...overrides,
+  },
+  loading: false,
+  refresh: () => undefined,
+});
+
+const createMockKnowledgeBase = (
+  overrides: Partial<UseKnowledgeBaseResult> = {}
+): UseKnowledgeBaseResult => ({
+  status: createMockStatus(),
+  isInstalling: false,
+  isWarmingUpModel: false,
+  isPolling: false,
+  install: jest.fn().mockResolvedValue(undefined),
+  warmupModel: jest.fn().mockResolvedValue(undefined),
+  ...overrides,
+});
+describe('ChangeKbModel', () => {
+  const modelOptions = [
+    {
+      key: ELSER_IN_EIS_INFERENCE_ID,
+      label: 'ELSER v2 (English-only)',
+      description:
+        'Focus on query meaning, not just keyword matching, using learned associations between terms. It delivers more relevant, context-aware results and works out of the box with no need for deep machine learning expertise.',
+    },
+    {
+      key: E5_SMALL_INFERENCE_ID,
+      label: 'E5-small (multilingual)',
+      description:
+        'E5 is an NLP model by Elastic designed to enhance multilingual semantic search by focusing on query context rather than keywords. E5-small is a cross-platform version compatible with different hardware configurations.',
+    },
+  ];
+
+  const setupMockGetModelOptions = (options = modelOptions) => {
+    mockGetModelOptions.mockReset();
+    mockGetModelOptions.mockReturnValue(options);
+  };
+
+  const renderComponent = (mockKb: any) => {
+    render(<ChangeKbModel knowledgeBase={mockKb} />);
+  };
+
+  beforeEach(() => {
+    setupMockGetModelOptions();
+  });
+
+  it('disables the button when selected model is the same as current and no redeployment needed', () => {
+    const mockKb = createMockKnowledgeBase();
+    renderComponent(mockKb);
+    const button = screen.getByTestId('observabilityAiAssistantKnowledgeBaseUpdateModelButton');
+    expect(button).toBeDisabled();
+  });
+
+  it('enables the button when a different model is selected', async () => {
+    const mockKb = createMockKnowledgeBase({
+      status: createMockStatus({ currentInferenceId: '.elser-2-elastic' }),
+    });
```

Comment:
I think you can use just `createMockKnowledgeBase()` here.
Because `.elser-2-elastic` is default `currentInferenceId` when you call `createMockStatus()`